app/OptionsIO/Sites/PhpVersion.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace App\OptionsIO\Sites;
+
+use App\Helpers\OptionsHelper;
+use App\OptionsIO\HasChoices;
+use App\OptionsIO\Option;
+
+class PhpVersion extends Option
+{
+    use HasChoices;
+
+    protected $default = '80';
+
+    protected string $promptValue = 'PHP Version';
+
+    public function getChoices(): array
+    {
+        return OptionsHelper::PHP_VERSIONS;
+    }
```

Comment:
Overall like the approach to Options. Any feedback I have is going to be on focused on making the process of writing new Options classes even more straightforward.

Any reason no abstract `HasChoices` out? `getChoices()` could always be present and just not used when not relevant. 

I'm also wondering if it would be easier if everything was defined in a getter method instead of overwriting the attributes directly.

Reply:
> Any reason no abstract HasChoices out? getChoices() could always be present and just not used when not relevant.

Since `askToSelect` has a different signature than the rest of the prompt types, I abstracted this out so that we can inspect the object when rendering the prompt in `HasOptionsIO` ([here](https://github.com/deliciousbrains/spinupwp-cli/pull/28/files#diff-d5ed48b1439efbdfdd07a8e155b218d6e20bcc53b24e657b3a9638dfbcf71beeR26)). Alternately, we could do an `if ( $this->prompType === 'askToSelect' )`.

> I'm also wondering if it would be easier if everything was defined in a getter method instead of overwriting the attributes directly.

I prefer defining the parameters vs getter because it's fewer lines of code (1 vs 3) and it feels Laravel-like (think of some of the Model params). I don't have a strong opinion here though.

Reply:
this thread is no longer relevant after latest refactor. resolving.